clients/client/src/main/java/org/projectnessie/client/http/UriBuilder.java

```diff
@@ -92,32 +89,35 @@ URI build() throws HttpClientException {
 
     if (uri.length() > 0) {
       Map<String, String> templates = new HashMap<>(templateValues);
-      // important note: this assumes an entire path component is the template. So /a/{b}/c will
-      // work but /a/b{b}/c will not.
-      Arrays.stream(uri.toString().split("/"))
-          .map(
-              p -> {
-                if (templates.containsKey(p)) {
-                  return encode(templates.remove(p));
-                } else if (p.contains("...")) {
-                  String[] diffs = p.split("\\.\\.\\.");
-                  if (diffs.length == 2) {
-                    if (templates.containsKey(diffs[0]) && templates.containsKey(diffs[1])) {
-                      String from = encode(templates.remove(diffs[0]));
-                      String to = encode(templates.remove(diffs[1]));
-                      return String.format("%s...%s", from, to);
-                    }
-                  }
-                }
-                return encode(p);
-              })
-          .forEach(
-              x -> {
-                if ('/' != uriBuilder.charAt(uriBuilder.length() - 1)) {
-                  uriBuilder.append('/');
-                }
-                uriBuilder.append(x);
-              });
+
+      StringBuilder pathElement = new StringBuilder();
+      StringBuilder name = new StringBuilder();
+      int l = uri.length();
+      for (int i = 0; i < l; i++) {
+        char c = uri.charAt(i);
+        if (c == '/') {
```

Comment:
Do we have to treat `/` chars in a special way? It looks like we can deal with `{` and `}` only. WDYT?

If we process `/` chars perhaps we can check that `{..}` groups do not intersect with path elements?.. However, I do not think it is necessary.

Reply:
I thought the same, but sadly the `encode` function also escapes `/`.
I didn't want to implement a custom encode function, but maybe we have to at one point, IDK.